### PR TITLE
Update to 5.00.0+domains

### PR DESCRIPTION
**Pull request by shubhamkumar13**
This PR updates/removes the dependencies and benchmarks needed to run `5.00.0+domains` variant with sandmark and sandmark-nightly

This PR removes the coq and irmin benchmarks because `coq-core` package seemed to have the following build failure
```bash
#=== ERROR while compiling coq-core.dev =======================================#
# context     2.0.7 | linux/x86_64 | ocaml-base-compiler.5.00.0+domains | pinned(https://github.com/ejgallego/coq/archive/refs/tags/multicore-2021-09-29.tar.gz)
# path        ~/sandmark/_opam/5.00.0+domains/.opam-switch/build/coq-core.dev
# command     ~/sandmark/_opam/5.00.0+domains/bin/dune build -p coq-core -j 27 @install
# exit-code   1
# env-file    ~/sandmark/_opam/log/coq-core-3876763-ce6ef9.env
# output-file ~/sandmark/_opam/log/coq-core-3876763-ce6ef9.out
### output ###
# Use the camlp-streams library instead.
# [...]
# (cd _build/default && /home/shubham/sandmark/_opam/5.00.0+domains/bin/ocamlc.opt -w -40 -rectypes -g -bin-annot -I engine/.engine.objs/byte -I engine/.engine.objs/public_cmi -I /home/shubham/sandmark/_opam/5.00.0+domains/lib/ocaml/threads -I boot/.boot.objs/public_cmi -I clib/.clib.objs/byte -I config/.config.objs/byte -I kernel/.kernel.objs/byte -I kernel/coqrun/.coqrun.objs/byte -I lib/.lib[...]
# File "engine/evarutil.ml", line 128, characters 10-18:
# 128 |   let m = E.create () in
#                 ^^^^^^^^
# Error: Unbound value E.create
#     ocamlopt engine/.engine.objs/native/evarutil.{cmx,o} (exit 2)
# (cd _build/default && /home/shubham/sandmark/_opam/5.00.0+domains/bin/ocamlopt.opt -w -40 -rectypes -g -O3 -unbox-closures -I engine/.engine.objs/byte -I engine/.engine.objs/native -I engine/.engine.objs/public_cmi -I /home/shubham/sandmark/_opam/5.00.0+domains/lib/ocaml/threads -I boot/.boot.objs/native -I boot/.boot.objs/public_cmi -I clib/.clib.objs/byte -I clib/.clib.objs/native -I config[...]
# File "engine/evarutil.ml", line 128, characters 10-18:
# 128 |   let m = E.create () in
#                 ^^^^^^^^
# Error: Unbound value E.create
```
The irmin benchmarks requires `ppxlib` which fails with `5.00.0+*` ocaml variants and hence is disabled in this PR

### CODE REVIEW

**Comment by kayceesrk** [comment]
The version on the multicore branch is also 5.00.

**Comment by kayceesrk** [comment]
The PR is removing the Coq and Irmin files. Is it not possible to keep the files in the same place, but only disable building them? Otherwise, it will not be apparent that we have only _disabled_ Coq and Irmin, but not _removed_.

File: Makefile [hunk 1]
```diff
@@ -31,7 +31,7 @@ WRAPPER = $(patsubst run_%,%,$(RUN_BENCH_TARGET))
 PACKAGES = \
 	cpdf conf-pkg-config conf-zlib bigstringaf decompress camlzip menhirLib \
 	menhir minilight base dune-private-libs dune-configurator camlimages \
-	yojson lwt zarith uuidm react ocplib-endian nbcodec checkseum \
+	yojson lwt zarith uuidm react ocplib-endian nbcodec checkseum ppxlib \
```
Comment (shakthimaan) [inline Makefile:34]: If Irmin benchmarks are removed, why do we need `ppxlib`?

**Comment by shakthimaan** [comment]
> The PR is removing the Coq and Irmin files. Is it not possible to keep the files in the same place, but only disable building them? 

Yes. Use a `sed` command to disable for 5.00. For example:
https://github.com/ocaml-bench/sandmark/blob/main/Makefile#L121

**Comment by kayceesrk** [comment]
The PR will also need to update https://github.com/ocaml-bench/sandmark#benchmarks-status.

File: Makefile (hunk 1 above)
Comment (shubhamkumar13) [reply Makefile:34]: removing ppxlib results in following benchmarks failing
```bash
File "benchmarks/chameneos/dune", line 5, characters 12-20:
5 |  (libraries lwt.unix))
                ^^^^^^^^
Error: Library "lwt.unix" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/decompress/dune", line 4, characters 41-54:
4 |   (libraries bigstringaf checkseum.ocaml decompress.zl))
                                             ^^^^^^^^^^^^^
Error: Library "decompress.zl" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/zarith/dune", line 1, characters 64-70:
1 | (executable (name zarith_fact) (modules zarith_fact) (libraries zarith))
                                                                    ^^^^^^
Error: Library "zarith" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/zarith/dune", line 2, characters 69-72:
2 | (executable (name zarith_fib) (modules zarith_fib) (libraries zarith num))
                                                                         ^^^
Error: Library "num" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/zarith/dune", line 3, characters 60-66:
3 | (executable (name zarith_pi) (modules zarith_pi) (libraries zarith))
                                                                ^^^^^^
Error: Library "zarith" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/zarith/dune", line 4, characters 69-72:
4 | (executable (name zarith_tak) (modules zarith_tak) (libraries zarith num))
                                                                         ^^^
Error: Library "num" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/benchmarksgame/dune", line 4, characters 34-40:
4 |               (libraries unix str zarith))
                                      ^^^^^^
Error: Library "zarith" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/thread-lwt/dune", line 5, characters 16-24:
5 |  (libraries lwt lwt.unix))
                    ^^^^^^^^
Error: Library "lwt.unix" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
File "benchmarks/valet/dune", line 5, characters 47-55:
5 |  (libraries uuidm ocplib-endian unix react lwt lwt.unix))
                                                   ^^^^^^^^
Error: Library "lwt.unix" not found.
Hint: try:
  dune external-lib-deps --missing --profile release --workspace ocaml-versions/.workspace.5.00.0+domains @buildbench
  ```

**Comment by kayceesrk** [comment]
Again, coq is “removed” but you probably want to “disable” it. By deleting those lines you are making it harder for a future contributor to figure out from scratch how to add coq; they only see the code of the latest commit and may not be aware that coq was present but disabled. Looking at the current code which has a number of lines deleted, they may find it harder to get back to a working state. 

Instead of deleting, a better approach may be to comment the deleted lines out? JSON doesn’t have comments but you may find other ways of disabling the coq benchmarks. For example, move the benchmarks that you want to disable to a different field whose name is “disabled_benchmarks”. The scripts won’t look at them.

**Comment by shakthimaan** [comment]
There is a fatal error when compiling base.v0.15.0. See: https://cloud.drone.io/ocaml-bench/sandmark/962/1/2
```
#=== ERROR while compiling base.v0.15.0 =======================================#
339s
674# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.5.00.0+trunk | pinned(git+https://github.com/janestreet/base.git#v0.15.0#2f23176f)
339s
675# path                 /drone/src/_opam/5.00.0+trunk/.opam-switch/build/base.v0.15.0
339s
676# command              /drone/src/_opam/5.00.0+trunk/bin/dune build -p base -j 47
339s
677# exit-code            1
339s
678# env-file             /drone/src/_opam/log/base-14687-f7741b.env
339s
679# output-file          /drone/src/_opam/log/base-14687-f7741b.out
339s
680### output ###
339s
681#          gen shadow-stdlib/src/shadow_stdlib.mli (exit 2)
339s
682# (cd _build/default/shadow-stdlib/src && ../gen/gen.exe -caml-cmi ../../compiler-stdlib/src/.caml.objs/caml.cmi ../../compiler-stdlib/src/.caml.objs/byte/caml.cmi -o shadow_stdlib.mli)
339s
683# Fatal error: exception Failure("no equivalent for symbol \"acosh\"")
```

File: Makefile [hunk 2]
```diff
@@ -88,11 +88,11 @@ _opam/%: _opam/opam-init/init.sh ocaml-versions/%.json setup_sys_dune
 	$(eval OCAML_RUN_PARAM     = $(shell jq -r '.runparams // empty' ocaml-versions/$*.json))
 	opam update
 	OCAMLRUNPARAM="$(OCAML_RUN_PARAM)" OCAMLCONFIGOPTION="$(OCAML_CONFIG_OPTION)" opam switch create --keep-build-dir --yes $* ocaml-base-compiler.$*
+	opam pin add -n --yes --switch $* base.v0.15.0 https://github.com/janestreet/base.git#v0.15.0
 	opam pin add -n --yes --switch $* eventlog-tools https://github.com/ocaml-multicore/eventlog-tools.git#multicore
+	opam pin add -n --yes --switch $* sexplib0.v0.15.0 https://github.com/shubhamkumar13/sexplib0.git#multicore
 	opam pin add -n --yes --switch $* orun orun/
 	opam pin add -n --yes --switch $* rungen rungen/
-	opam pin add -n --yes --switch $* coq-core https://github.com/ejgallego/coq/archive/refs/tags/multicore-2021-09-29.tar.gz
-	opam pin add -n --yes --switch $* coq-stdlib https://github.com/ejgallego/coq/archive/refs/tags/multicore-2021-09-29.tar.gz
```
Comment (shakthimaan) [inline Makefile:95]: Restore the above two lines, and just remove coq from `run_in_ci`?

File: Makefile [hunk 3]
```diff
@@ -108,12 +108,11 @@ blah:
 ocaml-versions/%.bench: check_url depend log_sandmark_hash ocaml-versions/%.json _opam/% .FORCE
 	$(eval ENVIRONMENT = $(shell jq -r '.wrappers[] | select(.name=="$(WRAPPER)") | .environment // empty' "$(RUN_CONFIG_JSON)" ))
 	@opam update
-	opam install --switch=$* --keep-build-dir --yes rungen orun coq-core coq-stdlib coq fraplib
```
Comment (shakthimaan) [inline Makefile:111]: Move `coq-core coq-stdlib coq fraplib` to the `*)` target in the following case expression.

**Comment by kayceesrk** [comment]
This looks like a known failure: https://github.com/janestreet/base/issues/125.

File: run_config.json [hunk 4]
```diff
@@ -1477,41 +1457,6 @@
...
-        }
-      ]
-    },
-    {
-      "executable": "coqc",
-      "name": "coq",
-      "tags": [
-        "gt_100s",
-        "macro_bench",
-        "run_in_ci"
```
Comment (shakthimaan) [inline run_config.json:1502]: Just remove `run_in_ci` temporarily for coq?

Comment (shubhamkumar13) [reply run_config.json:1502]: I removed `macro_bench` because sandmark-nightly uses `macro_bench` to run the benchmarks

**Comment by shubhamkumar13** [comment]
Thanks @shakthimaan @kayceesrk for reviewing the PR

**Pull request merged**